plugins/catalog-backend/src/actions/createQueryCatalogEntitiesAction.ts

```diff
@@ -0,0 +1,144 @@
+import { QueryEntitiesRequest } from '@backstage/catalog-client';
+
+export const createQueryCatalogEntitiesAction = ({
+  catalog,
+  actionsRegistry,
+}: {
+  catalog: CatalogService;
+  actionsRegistry: ActionsRegistryService;
+}) => {
+  actionsRegistry.register({
+    name: 'query-catalog-entities',
+    title: 'Query Catalog Entities',
+    attributes: {
+      destructive: false,
+      readOnly: true,
+      idempotent: true,
+    },
+    description: `
+This allows you to query multiple entities from the software catalog using filters.
+You can filter by entity properties like kind, namespace, metadata fields, and more.
+Using limit is recommended to avoid fetching too many entities at once. You can use the cursor returned in the response to fetch next set of entities.
+The action returns a list of entities matching the specified criteria.
+Fields are using dot notation, e.g. "kind", "metadata.name", "spec.type" in the filter, fields, orderFields and fullTextFilter.fields parameters.
+    `,
+    schema: {
+      input: z =>
+        z.object({
+          filter: z
+            .record(z.string())
+            .optional()
+            .describe(
+              'Filter criteria for querying entities. Keys are field paths and values are the expected values.',
+            ),
+          fields: z
+            .array(z.string())
+            .optional()
+            .describe(
+              'Specific fields to include in the response. If not provided, all fields are returned.',
+            ),
+          limit: z
+            .number()
+            .int()
+            .positive()
+            .optional()
+            .describe('Maximum number of entities to return.'),
+          offset: z
+            .number()
+            .int()
+            .min(0)
+            .optional()
+            .describe('Number of entities to skip before returning results.'),
+          orderFields: z
+            .object({
+              field: z.string().describe('Field to order by'),
+              order: z
+                .enum(['asc', 'desc'])
+                .optional()
+                .default('asc')
+                .describe('Sort order'),
+            })
+            .optional()
+            .describe('Ordering criteria for the results.'),
+          fullTextFilter: z
+            .object({
+              term: z.string().describe('Full text search term'),
+              fields: z
+                .array(z.string())
+                .optional()
+                .describe('Fields to search within'),
+            })
+            .optional()
+            .describe('Full text search criteria'),
+          cursor: z
+            .string()
+            .optional()
+            .describe(
+              'Cursor for pagination. This can be used only after first request with response containing a cursor',
+            ),
+        }),
+      output: z =>
+        z.object({
+          items: z
+            .array(z.object({}).passthrough())
```

Comment:
Not sure if it'll necessarily work when the filter returns only a subset of fields, but if you want, you can also specify the entity type with `z.custom<Entity>()` instead of `z.object`